api/v1beta1/hetznerbaremetalhost_webhook.go

```diff
@@ -43,28 +57,57 @@ func (host *HetznerBareMetalHost) Default() {
 
 //+kubebuilder:webhook:path=/validate-infrastructure-cluster-x-k8s-io-v1beta1-hetznerbaremetalhost,mutating=false,failurePolicy=fail,sideEffects=None,groups=infrastructure.cluster.x-k8s.io,resources=hetznerbaremetalhosts,verbs=create;update,versions=v1beta1,name=validation.hetznerbaremetalhost.infrastructure.cluster.x-k8s.io,admissionReviewVersions={v1,v1beta1}
 
-var _ webhook.Validator = &HetznerBareMetalHost{}
+var _ webhook.CustomValidator = &HetznerBareMetalHostWebhook{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
-func (host *HetznerBareMetalHost) ValidateCreate() (admission.Warnings, error) {
-	return nil, nil
+func (hw *HetznerBareMetalHostWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	host, ok := (obj).(*HetznerBareMetalHost)
+	if !ok {
+		return admission.Warnings{}, errors.NewBadRequest(fmt.Sprintf("expected HetznerBareMetalHost, but got %T", host))
+	}
+
+	var allErrs field.ErrorList
+
+	hetznerBareMetalHostList := &HetznerBareMetalHostList{}
+	if err := hw.client.List(ctx, hetznerBareMetalHostList, &ctrlClient.ListOptions{Namespace: host.Namespace}); err != nil {
+		return admission.Warnings{"could not verify that the host has a unique serverID"}, fmt.Errorf("failed to list hetznerBareMetalHost: %w", err)
```

Comment:
I meant to integrate the error in the warning message, not returning an error :see_no_evil: